include/envoy/network/listener.h

```diff
@@ -114,6 +114,51 @@ class ListenerCallbacks {
   virtual void onNewConnection(ConnectionPtr&& new_connection) PURE;
 };
 
+/**
+ * Utility struct that encapsulates the information from a udp socket's
+ * recvfrom/recvmmdg call.
+ *
+ * TODO (conqerAtapple): Maybe this belongs inside the UdpListenerCallbacks
+ * class.
+ */
+struct UdpData {
+  Address::InstanceConstSharedPtr local_address_;
+  Address::InstanceConstSharedPtr peer_address_;
+  Buffer::InstancePtr buffer_;
+};
+
+/**
+ * Udp listener callbacks.
+ */
+class UdpListenerCallbacks {
+public:
+  enum class ErrorCode { SYSCALL_ERROR, UNKNOWN_ERROR };
+
+  virtual ~UdpListenerCallbacks() = default;
+
+  /**
+   * Called whenever data is received by the underlying udp socket.
+   *
+   * @param data UdpData from the underlying socket.
+   */
+  virtual void onData(const UdpData& data) PURE;
+
+  /**
+   * Called when the underlying socket is ready for write.
+   *
+   * @param socket Underlying server socket for the listener.
+   */
+  virtual void onWriteReady(const Socket& socket) PURE;
+
+  /**
+   * Called when there is an error event.
+   *
+   * @param error_code ErrorCode for the error event.
+   * @param errno System error number.
+   */
+  virtual void onError(const ErrorCode& err_code, int err_no) PURE;
```

Comment:
super nitty, but can you use full variable names?
error_code error_number etc.?

Reply:
I think there's still a few of these outstanding.  Mind fixing up?

Reply:
I fixed them (latest commits) but looking at your comment again, I missed fixing `err_no` as you suggested. Will fix that.